RecoTracker/MkFitCore/src/Ice/IceFPU.h

```diff
@@ -0,0 +1,278 @@
+//----------------------------------------------------------------------
+/**
+ *	Contains FPU related code.
+ *	\file		IceFPU.h
+ *	\author		Pierre Terdiman
+ *	\date		April, 4, 2000
+ */
+//----------------------------------------------------------------------
+
+//----------------------------------------------------------------------
+// Include Guard
+#ifndef RecoTracker_MkFitCore_src_Ice_IceFPU_h
+#define RecoTracker_MkFitCore_src_Ice_IceFPU_h
+
+#define SIGN_BITMASK 0x80000000
+
+//! Integer representation of a floating-point value.
+#define IR(x) ((udword&)(x))
+
+//! Signed integer representation of a floating-point value.
+#define SIR(x) ((sdword&)(x))
+
+//! Absolute integer representation of a floating-point value
+#define AIR(x) (IR(x) & 0x7fffffff)
+
+//! Floating-point representation of an integer value.
+#define FR(x) ((float&)(x))
+
+//! Integer-based comparison of a floating point value.
+//! Don't use it blindly, it can be faster or slower than the FPU comparison, depends on the context.
+#define IS_NEGATIVE_FLOAT(x) (IR(x) & 0x80000000)
+
+//! Fast fabs for floating-point values. It just clears the sign bit.
+//! Don't use it blindy, it can be faster or slower than the FPU comparison, depends on the context.
+inline_ float FastFabs(float x) {
+  udword FloatBits = IR(x) & 0x7fffffff;
+  return FR(FloatBits);
+}
+
+#ifdef WIN32
+//! Fast square root for floating-point values.
+inline_ float FastSqrt(float square) {
+  float retval;
+
+  __asm {
+      mov             eax, square
+      sub             eax, 0x3F800000
+      sar             eax, 1
+      add             eax, 0x3F800000
+      mov             [retval], eax
+  }
+  return retval;
+}
+#endif
+
+//! Saturates positive to zero.
+inline_ float fsat(float f) {
+  udword y = (udword&)f & ~((sdword&)f >> 31);
+  return (float&)y;
+}
+
+//! Computes 1.0f / sqrtf(x).
+inline_ float frsqrt(float f) {
```

Comment:
my understanding is that this sort of tricks are not needed anymore with current standard library implementations.
Does this file need to be ported as-is?

Reply:
Use of Ice will be replaced by RecoTracker/MkFitCore/interface/binnor.h in a follow up PR.
The reason to not do it now is that bit-wise identity in the implementation is not guaranteed and we wanted to keep this PR technical (no changes in physics outputs).